src/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.CoreRT.cs

```diff
@@ -0,0 +1,207 @@
+                        // LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR is the only flag affecting absolute path. Don't OR the flags
+                        // unconditionally as all absolute path P/Invokes could then lose LOAD_WITH_ALTERED_SEARCH_PATH.
+                        flags |= dllImportSearchPathFlags;
+                    }
+
+                    ret = LocalLoadLibraryHelper(currLibNameVariation, flags, ref errorTracker);
+                    if (ret != IntPtr.Zero)
+                    {
+                        return ret;
+                    }
+                }
+                else if ((callingAssembly != null) && searchAssemblyDirectory)
+                {
+                    // Try to load the module alongside the assembly where the PInvoke was declared.
+                    // This only makes sense in dynamic scenarios (JIT/interpreter), so leaving this out for now.
+                }
+
+                ret = LocalLoadLibraryHelper(currLibNameVariation, dllImportSearchPathFlags, ref errorTracker);
+                if (ret != IntPtr.Zero)
+                {
+                    return ret;
+                }
+            }
+
+            return IntPtr.Zero;
+        }
+
+        private static IntPtr LoadFromPath(string libraryName, bool throwOnError)
+        {
+            LoadLibErrorTracker errorTracker = default;
+            IntPtr ret = LocalLoadLibraryHelper(libraryName, 0, ref errorTracker);
+            if (throwOnError && ret == IntPtr.Zero)
+            {
+                errorTracker.Throw(libraryName);
+            }
+
+            return ret;
+        }
+
+        private static IntPtr LocalLoadLibraryHelper(string libraryName, int flags, ref LoadLibErrorTracker errorTracker)
+        {
+#if PLATFORM_WINDOWS
+            IntPtr ret = Interop.mincore.LoadLibraryEx(libraryName, IntPtr.Zero, flags);
+            if (ret != IntPtr.Zero)
+            {
+                return ret;
+            }
+
+            int lastError = Marshal.GetLastWin32Error();
+            if (lastError != LoadLibErrorTracker.ERROR_INVALID_PARAMETER)
+            {
+                errorTracker.TrackErrorCode(lastError);
+            }
+
+            return ret;
+#else
+            IntPtr ret = IntPtr.Zero;
+            if (libraryName == null)
+            {
+                errorTracker.TrackErrorCode(LoadLibErrorTracker.ERROR_MOD_NOT_FOUND);
+            }
+            else if (libraryName == String.Empty)
+            {
+                errorTracker.TrackErrorCode(LoadLibErrorTracker.ERROR_INVALID_PARAMETER);
+            }
+            else
+            {
+                // TODO: FileDosToUnixPathA
+                ret = Interop.Sys.LoadLibrary(libraryName);
+                if (ret == IntPtr.Zero)
+                {
+                    errorTracker.TrackErrorCode(LoadLibErrorTracker.ERROR_MOD_NOT_FOUND);
+                }
+            }
+
+            return ret;
+#endif
+        }
+
+        private static void FreeLib(IntPtr handle)
```

Comment:
Nit: The usual naming pattern is to append `Core` to the internal helper. E.g. `FreeLibraryCore`